default.nix

```
@@ -0,0 +1,55 @@
+{
+  # crane flake
+  crane,
+  toolchainOverride ? null,
+  # standard
+  pkgs,
+  lib,
+  # dependencies
+  pkg-config,
+  llvm,
+  clang,
+  openssl,
+  makeWrapper,
+}:
+let
+  nativeBuildInputs = [
+    pkg-config
+    llvm
+    clang
+  ];
+  buildInputs = [
+    openssl
+  ];
+  toolchain = lib.defaultTo (
+    pkgs:
+    (pkgs.rust-bin.fromRustupToolchainFile ./rust-toolchain.toml).override {
+      extensions = [ "rust-src" ];
+    }
+  ) toolchainOverride;
+  craneLib = (crane.mkLib pkgs).overrideToolchain toolchain;
+  src = lib.cleanSourceWith {
+    src = craneLib.path ./.;
+    filter =
+      let
+        data-files = path: _type: builtins.match "^.*\\.(yaml|json|bin)$" path != null;
+      in
+      path: type: (data-files path type) || (craneLib.filterCargoSources path type);
+  };
+  commonArgsNoDeps = {
+    inherit src;
+    strictDeps = true;
+    inherit nativeBuildInputs buildInputs;
+  };
+
+  cargoArtifacts = craneLib.buildDepsOnly commonArgsNoDeps;
+  commonArgs = commonArgsNoDeps // {
+    inherit cargoArtifacts;
+  };
+in
+craneLib.buildPackage (
+  commonArgs
+)
```

```suggestion
craneLib.buildPackage commonArgs
```

Also, please run `nix run nixpkgs#nixfmt-rfc-style *.nix`